src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -31,6 +34,37 @@ const SHA224: i64 = 224;
 const SHA256: i64 = 256;
 const SHA384: i64 = 384;
 const SHA512: i64 = 512;
+//const AES_BLOCK_SIZE: i64 = 16;
+
+fn derive_key_mysql(input: &[u8], block_size: usize) -> Vec<u8> {
+    let mut output: Vec<u8> = vec![0; block_size];
+    let mut idx = 0;
+    for k in input {
+        if idx == block_size {
+            idx = 0;
+        }
+        output[idx] ^= k;
+        idx += 1;
+    }
+    output
+}
+
+fn get_key_size(key_size: usize) -> aes::KeySize {
+    match key_size {
+        16 => {
+            return aes::KeySize::KeySize128;
+        }
+        24 => {
+            return aes::KeySize::KeySize192;
+        }
+        32 => {
+            return aes::KeySize::KeySize256;
+        }
+        _ => {
+            panic!("not support key_size: {}", key_size);
```

Comment:
panic may not be not an ideal behaviour. (if using TryFrom, than it will be a normal Result, and will be ok)

Reply:
Has been modified to return an error